app/controllers/people_controller.rb

```diff
@@ -117,9 +115,28 @@ def create
 
   def update
     @person = Person.find(params[:id])
-    
+    channel_params = params[:person][:channels_attributes].values
+
+    unless channel_params.empty?
+        channel_params.each do |params|
+           if (params["_destroy"] == "1")
+              Channel.find(params["id"]).destroy
+           elsif (params.has_key?("id"))
+              params.reject! {|k,v| k=="_destroy"}
+              @person.channels.detect {|channel| channel.id == params["id"].to_i }.update_attributes(params)
+           else
+              params.reject! {|k,v| k=="_destroy"}
+              @channel = Channel.new(params)
+              @channel.person_id = @person.id
+              @channel.save
+           end
+        end
+    end
```

Comment:
Could you change your additions to use 2-space indents rather than this style of 3 and 4?